### DISCUSSION TITLE
Purge when offering a design system CSS

### DISCUSSION THREAD

**Discussion by melloc01** [tailwindlabs/tailwindcss#6117] [Ideas]
Hi everyone, I'm going to offer a "base CSS" for that is built with the company's Design System colors, sizes and spacing but it seems that I'll have to deliver the entire TailwindCSS, right? We're on a micro-frontend architecture and all teams should be able to use the same CSS.

It's not a big deal but I just wanted to share with you and check if I'm getting something wrong or something else.

Thanks!

**Comment by hacknug** [comment]
As long as you can build the styles on each of those enviroments, you should probably go with [`presets`](https://tailwindcss.com/docs/presets). This is what I'm using for one of my clients. It will let you have a common config file you use throughout all the other projects and avoid the need for a pre-built stylesheet with all of the possible classes your _users_ might need. Local overrides per project are also available with this method which is great for things like semantic colors.

If there's no way for you to build the styles per project taking advantage of JIT and/or purge, I'm afraid the prebuilt stylesheet approach might be your only option. Hopefully someone else has a better approach if this is the case.

**Reply by melloc01** [reply]
Thanks @hacknug, the presets feature is definitely a way to add intellisense to all projects. Actually I think I'm exporting the tailwind config  file in my project instead of creating a preset as we'll not support any customization on our plugins.

And actually, in our case it's probably best to offer the entire pre-built CSS for some reasons.

Creating one CSS per project will generate LOTS of duplicate CSS, I'm talking about 10 to 40 **different** front-end projects running at the same time on the browser. All of them with different repositories and build processes. 
We allow creating front-end code from the interface, thus we'd need tailwind classes for this _runtime/per environment_ project too.
Offering this complete CSS will save lots of questions and complexity that purging in every project would bring.

Summing up, less human complexity, less tooling complexity and in the long term less KBs, win win win.

Thanks for sharing your use case, have a great week!
